src/Stratis.Bitcoin/Utilities/JsonConverters/CoinJsonConverter.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Reflection;
+using NBitcoin;
+using NBitcoin.OpenAsset;
+using Newtonsoft.Json;
+
+namespace Stratis.Bitcoin.Utilities.JsonConverters
+{
+    /// <summary>
+    /// Converter used to convert an object implementing <see cref="ICoin"/> to and from JSON.
+    /// </summary>
+    /// <seealso cref="Newtonsoft.Json.JsonConverter" />
+    public class CoinJsonConverter : JsonConverter
+    {
+        public Network Network { get; set; }
+
+        public CoinJsonConverter(Network network)
+        {
+            this.Network = network;
+        }
+
+        /// <inheritdoc />
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(ICoin).GetTypeInfo().IsAssignableFrom(objectType.GetTypeInfo());
+        }
+
+        /// <inheritdoc />
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            return reader.TokenType == JsonToken.Null ? null : serializer.Deserialize<CoinJson>(reader).ToCoin();
+        }
+
+        /// <inheritdoc />
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            serializer.Serialize(writer, new CoinJson((ICoin)value, this.Network));
+        }
+    }
+
+    public class CoinJson
+    {
+        public uint256 TransactionId { get; set; }
+
+        public uint Index { get; set; }
+
+        public Money Value { get; set; }
+
+        public Script ScriptPubKey { get; set; }
+
+        public Script RedeemScript { get; set; }
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public BitcoinAssetId AssetId { get; set; }
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public long Quantity { get; set; }
+
+        public CoinJson()
+        {
+        }
+
+        public CoinJson(ICoin coin, Network network)
```

Comment:
Network isn't being used here.

Reply:
it's used here, line 83: `this.AssetId = cc.AssetId.GetWif(network);`

Reply:
OK cool!